pkg/provider/cloud/kubevirt/data_volume.go

```diff
@@ -32,28 +32,93 @@ import (
+}
+
+// reconcileLocalDataVolumes reconciles the local-disks from cluster.
+func reconcileLocalDataVolumes(ctx context.Context, cluster *kubermaticv1.Cluster, client ctrlruntimeclient.Client) error {
 	for _, d := range cluster.Spec.Cloud.Kubevirt.PreAllocatedDataVolumes {
 		dv, err := createPreAllocatedDataVolume(d, cluster.Status.NamespaceName)
 		if err != nil {
 			return err
 		}
-		dvCreator := []reconciling.NamedCDIv1beta1DataVolumeCreatorGetter{
-			dataVolumeCreator(dv),
+		if err := reconcileDataVolume(ctx, client, dv, cluster.Status.NamespaceName); err != nil {
+			return fmt.Errorf("failed to reconcile DataVolume: %w", err)
 		}
-		if err := reconciling.ReconcileCDIv1beta1DataVolumes(ctx, dvCreator, cluster.Status.NamespaceName, client); err != nil {
-			return fmt.Errorf("failed to reconcile Allocated DataVolume: %w", err)
+	}
+	return nil
+}
+
+// reconcilePreAllocatedDataVolume reconciles the preAllocated DataVolumes from standard VM images if cloning is enabled.
+func reconcilePreAllocatedDataVolume(ctx context.Context, dc *kubermaticv1.DatacenterSpecKubevirt, client ctrlruntimeclient.Client) error {
+	if dc.Images == nil || dc.Images.HTTP == nil || dc.Images.HTTP.ImageCloning == nil || !dc.Images.HTTP.ImageCloning.Enable {
+		return nil
+	}
+	existingDiskList := cdiv1beta1.DataVolumeList{}
+	listOption := ctrlruntimeclient.ListOptions{
+		Namespace: KubevirtDiskCacheNamespace,
+	}
+	if err := client.List(ctx, &existingDiskList, &listOption); ctrlruntimeclient.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	// pdvToBeRemoved contains info about pdv that are going to be removed.
+	pdvToBeRemoved := make(map[string]cdiv1beta1.DataVolume)
+	for _, dv := range existingDiskList.Items {
+		// only consider preAllocated DV and ignore custom-disks.
+		if _, exist := dv.Annotations[osAnnotationForCustomDisk]; !exist {
+			pdvToBeRemoved[dv.Name] = dv
+		}
+	}
+
+	for os, osVersion := range dc.Images.HTTP.OperatingSystems {
+		for version, url := range osVersion {
+			pdvCreateOrUpdate := kubermaticv1.PreAllocatedDataVolume{
+				Name:         fmt.Sprintf("%s-%s", os, version),
+				Annotations:  map[string]string{preAllocatedDVAnnotation: "false"},
+				URL:          url,
+				Size:         preAllocatedDVSize,
+				StorageClass: dc.Images.HTTP.ImageCloning.StorageClass,
+			}
+
+			dv, err := createPreAllocatedDataVolume(pdvCreateOrUpdate, KubevirtDiskCacheNamespace)
+			if err != nil {
+				return err
+			}
+			if err = reconcileDataVolume(ctx, client, dv, KubevirtDiskCacheNamespace); err != nil {
+				return fmt.Errorf("failed to reconcile DataVolume: %w", err)
+			}
+			// remove the created/updated pdv from the removal map.
+			delete(pdvToBeRemoved, pdvCreateOrUpdate.Name)
 		}
 	}
+
+	for _, dv := range pdvToBeRemoved {
+		if err := client.Delete(ctx, &dv); ctrlruntimeclient.IgnoreNotFound(err) != nil {
+			return fmt.Errorf("failed to remove Allocated DataVolume: %w", err)
+		}
+	}
+
 	return nil
 }
 
```

Comment:
maybe we could call this method `newDataVolume` as if I understand correctly, it's used for both DV (Pre-allocated and Custom).
`create` is a bit confusing as I first undesrstood that it would call some client.Create().